### PR TITLE
Check validity of 'become_method' values from code

### PR DESCRIPTION
**Pull request by ajinkyau**
Related: #3481

### CODE REVIEW

File: src/ansiblelint/rules/schema.py [hunk 1]
```diff
@@ -75,6 +77,49 @@ class ValidateSchemaRule(AnsibleLintRule):
...
+                    message=f"'become_method' must be one of the currently installed plugins: {', '.join(available_become_plugins_keys)}",
+                    lintable=file or Lintable(""),
+                    rule=ValidateSchemaRule(),
+                    details=ValidateSchemaRule.description,
+                    tag=f"schema[{file.kind}]",
+                ),
+            )
+
+        # check at task level
+        tasks = data.get("tasks", [])
```
Comment (ssbarnea) [inline src/ansiblelint/rules/schema.py:106]: This is not how you look inside tasks, we have `matchtask` method to implement which is called for each task within the rule. Look at parent implementation to see how is called.

Basically matchplay should never look inside `tasks` as we have a specific method that takes care of task level checks.

File: src/ansiblelint/rules/schema.py [hunk 2]
```diff
@@ -120,6 +165,9 @@ def matchyaml(self, file: Lintable) -> list[MatchError]:
                     tag=f"schema[{file.kind}]",
                 ),
             )
+
+        if not result:
+            result = super().matchyaml(file)
         return result
```
Comment (ssbarnea) [inline src/ansiblelint/rules/schema.py:173]: That is missing tests, it needs at least two tests, one with playbook and one with a task file.

Comment (ssbarnea) [reply src/ansiblelint/rules/schema.py:173]: I would also recommend making this easier to extend for other similar checks, where we want to verify that a specific field is one of list of generated values, maybe similar to:

```python
FIELD_CHECKS = {
"become_method": get_app().runtime.plugins.become.keys()
}
```


Also be sure that this is defined at module or class level and not inside the method being called.

The effective error message can be built from this dictionary, so we would not need to create new message for each field check that we add later.

**Review by ssbarnea — CHANGES_REQUESTED** [review]

File: src/ansiblelint/rules/schema.py (hunk 2 above)
Comment (ajinkyau) [reply src/ansiblelint/rules/schema.py:175]: updated

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**